plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftUnloadSplitManager.java

```diff
@@ -0,0 +1,195 @@
+
+import com.amazon.redshift.jdbc.RedshiftPreparedStatement;
+import com.amazon.redshift.util.RedshiftException;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import io.airlift.log.Logger;
+import io.trino.plugin.jdbc.ForRecordCursor;
+import io.trino.plugin.jdbc.JdbcClient;
+import io.trino.plugin.jdbc.JdbcColumnHandle;
+import io.trino.plugin.jdbc.JdbcProcedureHandle;
+import io.trino.plugin.jdbc.JdbcSplit;
+import io.trino.plugin.jdbc.JdbcSplitManager;
+import io.trino.plugin.jdbc.JdbcTableHandle;
+import io.trino.plugin.jdbc.PreparedQuery;
+import io.trino.plugin.jdbc.QueryBuilder;
+import io.trino.plugin.jdbc.logging.RemoteQueryModifier;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorSplitManager;
+import io.trino.spi.connector.ConnectorSplitSource;
+import io.trino.spi.connector.ConnectorTableHandle;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.Constraint;
+import io.trino.spi.connector.DynamicFilter;
+import io.trino.spi.connector.FixedSplitSource;
+import io.trino.spi.type.DecimalType;
+import io.trino.spi.type.TimeType;
+import io.trino.spi.type.VarbinaryType;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+
+import static io.trino.plugin.jdbc.JdbcDynamicFilteringSessionProperties.dynamicFilteringEnabled;
+import static io.trino.plugin.redshift.RedshiftSessionProperties.useUnload;
+import static java.util.Objects.requireNonNull;
+
+public class RedshiftUnloadSplitManager
+        implements ConnectorSplitManager
+{
+    private static final Logger log = Logger.get(RedshiftUnloadSplitManager.class);
+
+    private final JdbcClient jdbcClient;
+    private final QueryBuilder queryBuilder;
+    private final RemoteQueryModifier queryModifier;
+    private final JdbcSplitManager jdbcSplitManager;
+    private final String unloadLocation;
+    private final String unloadAuthorization;
+    private final ExecutorService executor;
+
+    @Inject
+    public RedshiftUnloadSplitManager(
+            JdbcClient jdbcClient,
+            QueryBuilder queryBuilder,
+            RemoteQueryModifier queryModifier,
+            JdbcSplitManager jdbcSplitManager,
+            RedshiftConfig redshiftConfig,
+            @ForRecordCursor ExecutorService executor)
+    {
+        this.jdbcClient = requireNonNull(jdbcClient, "jdbcClient is null");
+        this.queryBuilder = requireNonNull(queryBuilder, "queryBuilder is null");
+        this.queryModifier = requireNonNull(queryModifier, "queryModifier is null");
+        this.jdbcSplitManager = requireNonNull(jdbcSplitManager, "jdbcSplitManager is null");
+
+        this.unloadLocation = redshiftConfig.getUnloadLocation()
+                .map(location -> location.replaceAll("/$", "") + "/")
```

Comment:
We could disallow following `/` using `@Pattern` annotation in config class.

Reply:
added